activerecord/lib/active_record/connection_adapters/abstract/transaction.rb

```diff
@@ -620,6 +620,7 @@ def rollback_transaction(transaction = nil)
       end
 
       def within_new_transaction(isolation: nil, joinable: true)
+        isolation ||= @connection.pool.default_isolation_level
```

Comment:
I think to properly test this line ( specifically `||=` ) we should test this case : 
```ruby
 Tag.with_default_isolation_level(:read_committed) do
          assert_equal :read_committed, Tag.default_isolation_level
          Tag.transaction(isolation: :repeatable_read) do
            Tag.create!(name: "jon")
          end
        end
```
and that in the AS event it should match `BEGIN ISOLATION LEVEL REPEATABLE READ`

@byroot @kirs if you think this can be valuable, I can quickly create a PR for it